### DISCUSSION TITLE
DNS based proxy

### DISCUSSION THREAD

**Discussion by freakedenough** [atauenis/webone#68] [Ideas]
I wonder if there is a DNS or Gateway based HTTP 1.x proxy around which does the same as webone, but works via DNS instead.
There are many legacy browsers like MINUET which have no option to configure a proxy. Most of them can only configure a Gateway or DNS Server in their network setup.
https://en.wikipedia.org/wiki/Minnesota_Internet_Users_Essential_Tool 

What would be the solution for these? I think of a DNS based proxy:
- it has to run a HTTP and DNS server on a modern host
- client is configured to use the modern host as DNS
- once the modern host receives a request from a client, it will tell the client its own host IP as HTTP server IP
- in parallel it looks up the real server IP and fetches the requested document from the real server and
- forwards it to the client via HTTP 1.x via man-in-the-middle

**Comment by atauenis** [comment]
Early browsers don't send `Host` header, so DNS spoofing won't help.

For browsers without proxy support at all, there is a thing, currently called as "local mode". These browsers can access to WWW through WebOne by opening http://192.168.12.34:8080/http://www.example.com/index.htm .
